geo-types/src/line_string.rs

```diff
@@ -132,19 +138,52 @@ impl<'a, T: CoordNum> DoubleEndedIterator for PointsIter<'a, T> {
     }
 }
 
+/// A [Coordinate] iterator used by the `into_iter` method on a [LineString]
+#[derive(Debug)]
+pub struct CoordinatesIter<'a, T: CoordNum + 'a>(::std::slice::Iter<'a, Coordinate<T>>);
+
+impl<'a, T: CoordNum> Iterator for CoordinatesIter<'a, T> {
+    type Item = &'a Coordinate<T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.next()
+    }
+}
+
+impl<'a, T: CoordNum> DoubleEndedIterator for CoordinatesIter<'a, T> {
+    fn next_back(&mut self) -> Option<Self::Item> {
+        self.0.next_back()
+    }
+}
+
 impl<T: CoordNum> LineString<T> {
-    /// Return an iterator yielding the coordinates of a `LineString` as `Point`s
+    /// Return an iterator yielding the coordinates of a [LineString] as [Point]s
     pub fn points_iter(&self) -> PointsIter<T> {
         PointsIter(self.0.iter())
     }
 
-    /// Return the coordinates of a `LineString` as a `Vec` of `Point`s
+    /// Return an iterator yielding the members of a [LineString] as [Coordinate]s
+    pub fn iter(&self) -> impl Iterator<Item = &Coordinate<T>> {
+        self.0.iter()
+    }
+
+    /// Return an iterator yielding the coordinates of a [LineString] as mutable [Coordinate]s
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = &mut Coordinate<T>> {
+        self.0.iter_mut()
+    }
+
+    /// Return the coordinates of a [LineString] as a `Vec` of [Point]s
     pub fn into_points(self) -> Vec<Point<T>> {
         self.0.into_iter().map(Point).collect()
     }
 
-    /// Return an iterator yielding one `Line` for each line segment
-    /// in the `LineString`.
+    /// Return the coordinates of a [LineString] as a `Vec` of [Coordinate]s
+    pub fn into_coordinates(self) -> Vec<Coordinate<T>> {
+        self.0.into_iter().collect()
```

Comment:
Could this just return `.0`?

```suggestion
        self.0
```

Reply:
I'm fine with the name as-is, but another option is to name it `into_inner`, which returns the data the structure contains, which is a convention in the Rust standard library, and also our `Polygon#into_inner`

Reply:
Fixed `.0`🙈, and renamed.